### PR TITLE
Adds cancelation functionality

### PR DESCRIPTION
**Pull request by scentless-apprentice**

### CODE REVIEW

File: service/client_test.go [hunk 1]
```diff
@@ -134,3 +134,49 @@ func TestProviderStatusError(t *testing.T) {
 	assert.NotNil(resultJob.Status)
 	assert.EqualValues(resultJob.Status, "My status")
 }
+
+func TestCencelJob(t *testing.T) {
```
Comment (flavioribeiro) [inline service/client_test.go]: small typo, `CencelJob` instead of `CancelJob`

File: service/client_test.go [hunk 2]
```diff
@@ -134,3 +134,49 @@ func TestProviderStatusError(t *testing.T) {
...
+	}
+	client.DB.StoreJob(job)
+	resultJob, _ := client.GetJob(job.ID)
+
+	canceled, err := client.CancelJob(resultJob.ID)
+	assert.Nil(err)
+	assert.True(canceled)
+}
+
+func TestCencelJobDone(t *testing.T) {
```
Comment (flavioribeiro) [inline service/client_test.go]: small typo, `CencelJob` instead of `CancelJob`

File: service/client_test.go [hunk 3]
```diff
@@ -134,3 +134,49 @@ func TestProviderStatusError(t *testing.T) {
...
+	}
+	client.DB.StoreJob(job)
+	resultJob, _ := client.GetJob(job.ID)
+
+	canceled, err := client.CancelJob(resultJob.ID)
+	assert.Nil(err)
+	assert.False(canceled)
+}
+
+func TestCencelJobDoesntExist(t *testing.T) {
```
Comment (flavioribeiro) [inline service/client_test.go]: small typo, `CencelJob` instead of `CancelJob`

File: service/job.go [hunk 4]
```diff
@@ -83,6 +83,19 @@ func (s *CaptionsService) GetJob(r *http.Request) (int, interface{}, error) {
...
 
+// CancelJob cancels a given Job by its ID
+func (s *CaptionsService) CancelJob(r *http.Request) (int, interface{}, error) {
+	id := web.Vars(r)["id"]
+	canceled, err := s.client.CancelJob(id)
+	if err != nil {
+		return http.StatusNotFound, nil, captionsError{err.Error()}
+	}
+	if !canceled {
+		return http.StatusConflict, nil, captionsError{"Cannot cancel a job that is already done"}
```
Comment (flavioribeiro) [inline service/job.go]: nice use of `StatusConflict` 👍

**Comment by flavioribeiro** [comment]
thanks!

File: service/job.go [hunk 5]
```diff
@@ -83,6 +83,19 @@ func (s *CaptionsService) GetJob(r *http.Request) (int, interface{}, error) {
 	return http.StatusOK, job, nil
 }
 
+// CancelJob cancels a given Job by its ID
+func (s *CaptionsService) CancelJob(r *http.Request) (int, interface{}, error) {
+	id := web.Vars(r)["id"]
+	canceled, err := s.client.CancelJob(id)
+	if err != nil {
+		return http.StatusNotFound, nil, captionsError{err.Error()}
```
Comment (markx) [inline service/job.go]: This `err` might be from `c.DB.UpdateJob(jobID, job)`, in which case returning `404` could be confusing.

**Pull request merged**